zebra/zebra_nhg.c

```diff
@@ -525,10 +525,19 @@ static bool nhg_compare_nexthops(const struct nexthop *nh1,
 	    != CHECK_FLAG(nh2->flags, NEXTHOP_FLAG_ACTIVE))
 		return false;
 
-	if (!nexthop_same(nh1, nh2))
-		return false;
+	bool same = nexthop_same(nh1, nh2);
 
-	return true;
+	/*
+	 * In case of both nexthops having a resolved nexthop, check if both those
```

Comment:
I'm still not clear about this change.
This function is used to compare the nexthops that are forming an nhg - the list of nexthops in a route via zapi, or a daemon-owned nhg (also via zapi). this is different from the complete walk of the fully-resolved set of nexthops used in nexthop_group_hash(), for example.